sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvoker.java

```diff
@@ -52,6 +52,10 @@
   DoFn.ProcessContinuation invokeProcessElement(
       DoFn<InputT, OutputT>.ProcessContext c, DoFn.ExtraContextFactory<InputT, OutputT> extra);
 
+  /** Invoke the appropriate {@link DoFn.OnTimer} method on the bound {@link DoFn}. */
+  void invokeOnTimer(
+      String timerId, DoFn.ExtraContextFactory<InputT, OutputT> extra);
```

Comment:
Note on this signature: it is my intent to put _all_ args into the `ExtraContextFactory`, including the `OnTimerContext` that allows output and side inputs. Eventually, also `ProcessContext`. It can then be entirely up to `DoFnSignatures` to validate it, as doing otherwise is redundant and less flexible.